src/w3d/math/matrix4.h

```diff
@@ -354,7 +354,7 @@ class Matrix4
         out->X = (A[0][0] * in.X + A[0][1] * in.Y + A[0][2] * in.Z + A[0][3]);
         out->Y = (A[1][0] * in.X + A[1][1] * in.Y + A[1][2] * in.Z + A[1][3]);
         out->Z = (A[2][0] * in.X + A[2][1] * in.Y + A[2][2] * in.Z + A[2][3]);
-        out->W = 1.0f;
+        out->W = (A[3][0] * in.X + A[3][1] * in.Y + A[3][2] * in.Z + A[3][3]);
```

Comment:
I cannot recall why this was changed or why this would be correct, so I am going to revert this in this change.